text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1015 @@
+message Stacktrace {
+  repeated uint32 location_indices = 1;
+}
+
+// AggregationTemporality defines how a profile aggregator reports aggregated
+// values. It describes how those values relate to the time interval over
+// which they are aggregated.
+enum AggregationTemporality {
+  AGGREGATION_TEMPORALITY_UNSPECIFIED = 0; // Temporality unspecified.
+  AGGREGATION_TEMPORALITY_DELTA = 1;       // Delta aggregation over time.
+  AGGREGATION_TEMPORALITY_CUMULATIVE = 2;  // Cumulative aggregation over time.
+}
+
+// Represents a single profile type. It implicitly creates a connection between Stacktraces, Links, AttributeSets, values and timestamps. The connection is based on the order of the elements in the corresponding lists. This implicit connection creates an ephemeral structure called Sample. The length of reference lists must be the same. It is acceptable however for timestamps, links and attribute set lists to be empty. It is not acceptable for stacktrace or values lists to be empty.
+message ProfileType {
+  // aggregation_temporality describes if the aggregator reports delta changes
+  // since last report time, or cumulative changes since a fixed start time.
+  AggregationTemporality aggregation_temporality = 1;
+
+  // Profiler sample rate in Hz. This parameter indicates the frequency at which samples are collected, specifically for CPU profiles. Common values are 99 or 100. [Optional].
+  uint64 sample_rate = 2;
+
+  // Index into the string table for the type of the sample. Example values are "cpu", "alloc_objects", "alloc_bytes", "block_contentions". Full list is defined in https://github.com/open-telemetry/semantic-conventions
+  uint32 type_index = 3;
+
+  // Index into the string table for the unit of the sample. Example values are "ms", "ns", "samples", "bytes". Full list is defined in https://github.com/open-telemetry/semantic-conventions
+  uint32 unit_index = 4;
+
+  // List of indices referring to Stacktraces in the Profile's stacktrace table.
+  repeated uint32 stacktrace_indices = 10;
+
+  // List of indices referring to Links in the Profile's link table. Each link corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length. [Optional]
+  repeated uint32 link_indices = 11;
+
+  // List of indices referring to AttributeSets in the Profile's attribute set table. Each attribute set corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length. [Optional]
+  repeated uint32 attribute_set_indices = 12;
+
+  // List of values. Each value corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length.
+  repeated int64 values = 13;
```

Comment:
I think it'd be great to have an example of encoding a CPU profile containing time stamps. Since CPU profile samples usually carry the same value weight (10ms), would this be just a long list of repeating this value?